scripts/validate.mjs

```diff
@@ -64,3 +100,34 @@ function forEachFile(directoryPath, callback) {
 function readJson(filePath) {
   return JSON.parse(readFileSync(filePath, 'utf8'))
 }
+
+/**
+ * Iterates over each properties of objects specified in the provided JSON schema.
+ */
+function forEachObjectProperty(schema, callback) {
+  if (Array.isArray(schema)) {
+    // traverse arrays
+    for (const value of schema) {
+      forEachObjectProperty(value, callback)
+    }
+  } else if (typeof schema === 'object' && schema !== null) {
+    // traverse objects
+    for (const value of Object.values(schema)) {
+      forEachObjectProperty(value, callback)
+    }
+
+    if (schema.type === 'object' && schema.properties) {
```

Comment:
**question:**
when `properties` exists in `schema`, it can only be an object? shouldn't you check if `properties` is different of `null` in case of?

Reply:
Yes `properties` is always an object: https://json-schema.org/understanding-json-schema/reference/object#properties